addons/mail/static/src/components/emoji_grid_view/emoji_grid_view.xml

```diff
@@ -3,7 +3,8 @@
 
     <t t-name="mail.EmojiGridView" owl="1">
     <div class="o_EmojiGridView o_EmojiGridView_container" t-attf-style="--rowHeight: {{emojiGridView.rowHeight}}px; --itemWidth: {{emojiGridView.itemWidth}}px; height: {{emojiGridView.height}}; width: {{emojiGridView.width}};" t-on-scroll="emojiGridView.onScroll" t-ref="containerRef">
-        <div class="o_EmojiGridView_list" t-attf-style="height: {{emojiGridView.listHeight}}px; min-height: {{emojiGridView.listHeight}}px;" t-ref="listRef">
+        <EmojiGridSearchNoContentView t-if="emojiGridView.searchNoContentView" record="emojiGridView.searchNoContentView"/>
+        <div t-if="!emojiGridView.searchNoContentView" class="o_EmojiGridView_list" t-attf-style="height: {{emojiGridView.listHeight}}px; min-height: {{emojiGridView.listHeight}}px;" t-ref="listRef">
```

Comment:
`EmojiGridView` is poorly named, but it's clearly the part of emoji picker below search bar and category bar.
It's actually `EmojiPickerContentView` or something like that.

In that regard, it makes total sense to have `searchNoContentView` here, because we want to keep search bar and category while showing this visual (and ensuring it shows error content with same layout as when there are some content)

So the placement of `NoSearchView` is good. The naming `EmojiGrid` is the problem.

Reply:
I agree, renaming `EmojiGridView` seems to solve the modelling problems I was thinking of. Do you plan to do so in the next few PRs?

Reply:
Yes will do. Note however v16 is coming very soon and it's a stable version, so it becomes crucial to treat master a bit like v16 for the time being. That's why it's more appropriate to treat fixes like stable fixes right now.